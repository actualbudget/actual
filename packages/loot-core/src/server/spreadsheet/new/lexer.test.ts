import lex from './lexer';

function getTokens(tokens) {
  const toks = [];
  while (!tokens.is_finished()) {
    toks.push(tokens.nextToken());
  }
  return toks;
}

test('lexer basic', () => {
  const tokens = lex(`
    =x !=~ 4
  `);

  expect(getTokens(tokens)).toMatchSnapshot();
});
