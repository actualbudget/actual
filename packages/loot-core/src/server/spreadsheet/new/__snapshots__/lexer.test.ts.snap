// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lexer basic 1`] = `
Array [
  Object {
    "colno": 0,
    "lineno": 0,
    "type": "whitespace",
    "value": "
    ",
  },
  Object {
    "colno": 5,
    "lineno": 1,
    "type": "symbol",
    "value": "x",
  },
  Object {
    "colno": 6,
    "lineno": 1,
    "type": "whitespace",
    "value": " ",
  },
  Object {
    "colno": 7,
    "lineno": 1,
    "type": "operator",
    "value": "!=~",
  },
  Object {
    "colno": 10,
    "lineno": 1,
    "type": "whitespace",
    "value": " ",
  },
  Object {
    "colno": 11,
    "lineno": 1,
    "type": "int",
    "value": "4",
  },
  Object {
    "colno": 12,
    "lineno": 1,
    "type": "whitespace",
    "value": "
  ",
  },
]
`;
